src/julia.h

```diff
@@ -624,30 +624,30 @@ static inline void gc_wb_back(void *ptr) // ptr isa jl_value_t*
 #define jl_gc_n_preserved_values() (0)
 
 #define allocb(nb)    malloc(nb)
-DLLEXPORT jl_value_t *allocobj(size_t sz);
-STATIC_INLINE jl_value_t *alloc_1w() { return allocobj(1*sizeof(void*)); }
-STATIC_INLINE jl_value_t *alloc_2w() { return allocobj(2*sizeof(void*)); }
-STATIC_INLINE jl_value_t *alloc_3w() { return allocobj(3*sizeof(void*)); }
+DLLEXPORT jl_value_t *jl_gc_allocobj(size_t sz);
+STATIC_INLINE jl_value_t *jl_gc_alloc_1w() { return jl_gc_allocobj(1*sizeof(void*)); }
+STATIC_INLINE jl_value_t *jl_gc_alloc_2w() { return jl_gc_allocobj(2*sizeof(void*)); }
+STATIC_INLINE jl_value_t *jl_gc_alloc_3w() { return jl_gc_allocobj(3*sizeof(void*)); }
 
 DLLEXPORT void jl_gc_add_finalizer(jl_value_t *v, jl_function_t *f);
 
-int64_t diff_gc_total_bytes(void);
-#define sync_gc_total_bytes()
+int64_t jl_gc_diff_total_bytes(void);
+#define jl_gc_sync_total_bytes()
 #define jl_gc_collect(arg);
 #define jl_gc_enable(on) (0)
 #define jl_gc_is_enabled() (0)
 #define jl_gc_track_malloced_array(a)
 #define jl_gc_count_allocd(sz)
 
-#define gc_wb_binding(bnd, val)
-#define gc_wb(parent, ptr)
-#define gc_wb_buf(parent, bufptr)
-#define gc_wb_back(ptr)
+#define jl_gc_wb_binding(bnd, val)
+#define jl_gc_wb(parent, ptr)
+#define jl_gc_wb_buf(parent, bufptr)
+#define jl_gc_wb_back(ptr)
 
 #endif
 
-DLLEXPORT void *jl_gc_managed_malloc(size_t sz);
-DLLEXPORT void *jl_gc_managed_realloc(void *d, size_t sz, size_t oldsz, int isaligned, jl_value_t* owner);
+void *jl_gc_managed_malloc(size_t sz);
```

Comment:
This still has `DLLEXPORT` at the implementation in `gc.c`, at least with MSVC the linkage needs to be consistent. I'd err on the side of leaving this exported, since someone saw the need to put `DLLEXPORT` on it in the first place.